content/en/docs/overview/singleton-sdk/Angular Client Introduction/i18n-service.md

```diff
@@ -0,0 +1,144 @@
+---
+title: "I18n Service"
+date: 2019-09-24T20:08:31+08:00
+draft: false
+weight: 30
+---
+
+
+#### **Overview**
+
+I18n service provides APIs which are supposed to be used in JavaScript module to get supported languages & regions, and format locale-sensitive data.
+
+![i18n-service-1](https://github.com/zmengjiao/singleton/raw/website/content/en/images/i18n-service/i18n-service-1.png)
+
+
+#### **Supported Language & Region APIs**
+
+##### **getSupportedLanguages**
+
+Get the supported languages which are available in singleton service for the specific product.
+
+```
+getSupportedLanguages(): Promise<languagesList>;
+
+```
+
+##### **getSupportedRegions**
+
+Get the supported regions from singleton service, the supported region list should be unified in all products.
+
+```
+getSupportedRegions(languageTag: string): Promise<regionsList>;
+
+```
+
+|  Parameter  |  Type  |                         Description                          |
+| :---------: | :----: | :----------------------------------------------------------: |
+| languageTag | String | Get supported region list with the localized display name by the 'languageTag'. |
+
+
+##### **Example Code**
+
+![i18n-service-2](https://github.com/zmengjiao/singleton/raw/website/content/en/images/i18n-service/i18n-service-2.png)
+
+
+
+#### **Data Formatting APIs**
+
+Data formatting APIs are locale-sensitive, they return depends on the locale. So when using these formatting APIs in JavaScript module, locale is a required parameter which can be fetched through the locale service by synchronous or asynchronous API that depends on the way of i18n data initialization. if the i18n data was loaded in synchronous by 'APP_INITIALIZER' token, the synchronous API can be used to get locale directly. Otherwise, the stream API is a right way to get locale through the subscription. For more details, please refer to the sessions 'Data initialization' and 'Locale Management'.
+
+
+##### **Format date**
+
+Formats a date according to the given options and locale.
+
+
+```
+formatDate(value: any, pattern: string, locale?: string, timezone?: string): string;
+
+```
+
+| Parameter |  Type  |                         Description                          |
+| :-------: | :----: | :----------------------------------------------------------: |
+|   value   |  any   | a date object or a number (milliseconds since UTC epoch) or an ISO string (https://www.w3.org/TR/NOTE-datetime). |
+|  pattern  | string | indicates which date/time components to include. The format can be predefined as shown [here](https://ngx.eng.vmware.com/@vmw/ngx-vip/date-format). Default is 'mediumDate'. |
+|  locale   | string | The locale is only required in async mode for non-blocking loading. Please refer to the [stream API](https://ngx.eng.vmware.com/@vmw/ngx-vip/locale-management/documentation#Asynchronous-API) usage. |
```

Comment:
URL issue [here] and [stream API]  as above

Reply:
I fixed them all.